force-app/main/default/classes/NextStepFuture.cls

```diff
@@ -58,19 +78,20 @@ public with sharing class NextStepFuture {
 
 
     // Helper method to update role name
-    private static void updateUserRoleName(UserRole[] roles){
+    @future
+    public static void updateUserRoleName(List<String> roleStrings){
+        List<UserRole> initialRole = [SELECT Id, Name FROM UserRole WHERE Id IN :roleStrings];
         UserRole role;
-        if (roles.isEmpty()) {
+        if (initialRole.isEmpty()) {
             role = new UserRole();
             role.portalType = 'None'; 
             role.CaseAccessForAccountOwner = 'Edit'; 
             // role.OpportunityAccessForAccountOwner = 'Edit'; //This may need to be uncommented depending on your orgs Organization Wide Defaults(OWD) settings. If you are receiving FIELD_INTEGRITY_EXCEPTION, field integrity exception: unknown (Opportunity access level below organization default): [unknown], uncomment this line.
         } else {
-            role = roles[0];
+            role = initialRole[0];
```

Comment:
-Not a problem but I'm curious why you changed the method signature here to take the roleIds rather than the role records. If it was because of the Mixed DML error, that doesn't really make a difference. Making sure that your DML is inside of an @future method like you did is the real magic